staging/src/k8s.io/kubectl/pkg/cmd/set/set_serviceaccount.go

```diff
@@ -204,6 +204,11 @@ func (o *SetServiceAccountOptions) Run() error {
 		info := patch.Info
 		name := info.ObjectName()
 		if patch.Err != nil {
+			if ee, ok := patch.Err.(polymorphichelpers.ResourceNotHavePodTemplate); ok {
+				klog.Warning(ee)
+				continue
```

Comment:
Now I think about it more, I am not sure if this is the correct way to go.

What if the user only has `Service` object in the yaml file?
Let's say the user has `foo.yaml`
```
apiVersion: v1
kind: Service
metadata:
  name: my-service
spec:
  type: NodePort
  selector:
    pod: my-pod
  ports:
  - port: 1234
    targetPort: 1234
```
and use the command 
```
./kubectl set image -f foo.yaml my-pod=nginx:1.9.1 -o yaml
```
The output is returning a warning but the error code is actually showing 0 , so there is a gray area whether this command succeeds or not.

Of course this is just an extreme case, so I need more inputs here,let me see if I can get some folks chip in.

Reply:
Thanks for review @lauchokyip. You mentioned a good point. I agree that if there is any other resource successfully set, this should be error instead just warning. 

For the example you gave, it'll throw error.

But for this;

```
kind: pod
metadata:
  name: my-pod
spec:
  containers:
  - name: my-pod
    image: foo:latest
---
kind: service
metadata:
  name: my-service
spec:
  type: NodePort
  selector:
    pod: my-pod
  ports:
  - port: 1234
    targetPort: 1234
```

After warning user about `pod` resource, it will successfully set to `service`. What do you think?